maxim/logger/components/generation.py

```
@@ -223,33 +267,76 @@ def set_provider(self, provider: str):
         if "content" not in message or "role" not in message:
             scribe().error(
                 "[MaximSDK] Invalid message. Must have 'content' and 'role' keys. We are skipping adding this message."
             )
             return
+        messages, attachments = parse_attachments_from_messages([message])
+        if len(attachments) > 0:
+            for attachment in attachments:
+                Generation.add_attachment_(writer, id, attachment)
         BaseContainer._commit_(
-            writer, Entity.GENERATION, id, "update", {"messages": [message]}
+            writer, Entity.GENERATION, id, "update", {"messages": messages}
```

Ruff (0.11.9)

Trailing comma missing

Add trailing comma

(COM812)